scripts/ci/lib.sh

```diff
@@ -202,6 +202,103 @@ push_main_image_set() {
     done
 }
 
+push_operator_image_set() {
+    info "Pushing stackrox-operator, stackrox-operator-bundle and stackrox-operator-index images"
+
+    if [[ "$#" -ne 2 ]]; then
+        die "missing arg. usage: push_operator_image_set <push_context> <brand>"
+    fi
+
+    local push_context="$1"
+    local brand="$2"
+
+    local operator_image_set=("stackrox-operator" "stackrox-operator-bundle" "stackrox-operator-index")
+    if is_OPENSHIFT_CI; then
+        local operator_image_srcs=("$OPERATOR_IMAGE" "$OPERATOR_BUNDLE_IMAGE" "$OPERATOR_BUNDLE_INDEX_MAGE")
+        oc registry login
+    fi
+
+    _push_operator_image_set() {
+        local registry="$1"
+        local tag="$2"
+
+        local v
+        for image in "${operator_image_set[@]}"; do
+            if [[ "${image}" != "stackrox-operator" ]]; then
+                # Only the bundle and index image tags have the v prefix.
+                v="v"
+            else
+                v=""
+            fi
+            "$SCRIPTS_ROOT/scripts/ci/push-as-manifest-list.sh" "${registry}/${image}:${v}${tag}" | cat
+        done
+    }
+
+    _tag_operator_image_set() {
+        local local_tag="$1"
+        local registry="$2"
+        local remote_tag="$3"
+
+        local v
+        for image in "${operator_image_set[@]}"; do
+            if [[ "${image}" != "stackrox-operator" ]]; then
+                # Only the bundle and index image tags have the v prefix.
+                v="v"
+            else
+                v=""
+            fi
+            docker tag "stackrox/${image}:${local_tag}" "${registry}/${image}:${v}${remote_tag}"
+        done
+    }
+
+    _mirror_operator_image_set() {
+        local registry="$1"
+        local tag="$2"
+
+        local idx=0
+        local v
+        for image in "${operator_image_set[@]}"; do
+            if [[ "${image}" != "stackrox-operator" ]]; then
+                # Only the bundle and index image tags have the v prefix.
+                v="v"
+            else
+                v=""
+            fi
+            oc image mirror "${operator_image_srcs[$idx]}" "${registry}/${image}:${v}${tag}"
+            (( idx++ )) || true
+        done
+    }
+
+    if [[ "$brand" == "STACKROX_BRANDING" ]]; then
+        local destination_registries=("quay.io/stackrox-io")
+    elif [[ "$brand" == "RHACS_BRANDING" ]]; then
+        local destination_registries=("quay.io/rhacs-eng")
```

Comment:
Does it make sense to have the registry repos configurable by env variables?

Reply:
Well, we need credentials for pushing and these are passed in via the CI vault and they are, of course, for a fixed registry. So, in any case, changing the registry would not be simply a matter of changing some registry env variable.

What I attempted the other day was also putting the registry right next to username & password into the vault, so they form a group, like:
```
IMAGE_PUSH_REGISTRY_NAME
IMAGE_PUSH_REGISTRY_USERNAME
IMAGE_PUSH_REGISTRY_PASSWORD
```

But the downside of this is that whenever you would like to output someothing like "Pushing $IMG to $REGISTRY", the registry will be redacted by OSCI. So, this is also not ideal. I would say, under these circumstances it's not an issue to hardcode the registry.

These are just my 2c.

Reply:
What I [alluded (probably not strongly enough) to in the commit description](https://github.com/stackrox/stackrox/pull/2471/commits/5ae1bc4a5ef1a102df992e03b5246bbd8412be43) is that this whole function is an almost exact copy of `push_main_image_set`, so I'm just being consistent with what the other does.

Unfortunately due to bash limitations **and** the inconsistent versioning scheme of our images, it would be a bit arcane to try and reuse code between them.

We can think about improving both of them, but I'd consider this to be out of scope for this PR.